### PR TITLE
improvement(header/p2p): Implement context deadline on requests to store inside server

### PR DESCRIPTION
**Pull request by renaynay**
This PR introduces server-side context deadlines for **all** requests down to store with a config parameter for the server `RequestTimeout`

### CODE REVIEW

**Review by vgonkivs — COMMENTED** [review]
Could you update this test please [#L87](https://github.com/celestiaorg/celestia-node/blob/97c07a44c5966af738c138e29be6e4a65c2db481/nodebuilder/header/module_test.go#L87) ?

**Review by Wondertan — COMMENTED** [review]
LGTM. Pls also do Slava's suggestion

**Review by Wondertan — DISMISSED** [review]

**Review by vgonkivs — APPROVED** [review]

File: header/p2p/options.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ type ServerParameters struct {
 	ReadDeadline time.Duration
 	// MaxRequestSize defines the max amount of headers that can be handled at once.
 	MaxRequestSize uint64
+	// RequestTimeout defines the deadline for server-side requests.
+	RequestTimeout time.Duration
```
Comment (Wondertan) [inline header/p2p/options.go:27]: Should we call it ServeTimeout?

Comment (Wondertan) [reply header/p2p/options.go:27]: Like the time given to serve a request

File: header/p2p/options.go [hunk 2]
```diff
@@ -82,6 +88,17 @@ func WithMaxRequestSize[T parameters](size uint64) Option[T] {
 	}
 }
 
+// WithRequestTimeout is a functional option that configures the
+// `ServeTimeout` parameter.
+func WithRequestTimeout[T parameters](timeout time.Duration) Option[T] {
```
Comment (Wondertan) [inline header/p2p/options.go:93]: Also option field

**Review by vgonkivs — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**